src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassaymetadata/CorrelationAssayMetadataPlugin.java

```diff
@@ -0,0 +1,151 @@
+package org.veupathdb.service.eda.compute.plugins.correlationassaymetadata;
+
+import org.gusdb.fgputil.ListBuilder;
+import org.jetbrains.annotations.NotNull;
+import org.veupathdb.service.eda.common.client.spec.StreamSpec;
+import org.veupathdb.service.eda.common.model.EntityDef;
+import org.veupathdb.service.eda.common.model.ReferenceMetadata;
+import org.veupathdb.service.eda.common.model.VariableDef;
+import org.veupathdb.service.eda.common.plugin.util.PluginUtil;
+import org.veupathdb.service.eda.compute.RServe;
+import org.veupathdb.service.eda.compute.plugins.AbstractPlugin;
+import org.veupathdb.service.eda.compute.plugins.PluginContext;
+import org.veupathdb.service.eda.generated.model.CorrelationComputeConfig;
+import org.veupathdb.service.eda.generated.model.CorrelationPluginRequest;
+import org.veupathdb.service.eda.generated.model.VariableSpec;
+import org.veupathdb.service.eda.generated.model.APIVariableDataShape;
+import org.veupathdb.service.eda.generated.model.CollectionSpec;
+
+import java.io.InputStream;
+import java.util.stream.Stream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.veupathdb.service.eda.common.plugin.util.PluginUtil.singleQuote;
+
+public class CorrelationAssayMetadataPlugin extends AbstractPlugin<CorrelationPluginRequest, CorrelationComputeConfig> {
+
+  private static final String INPUT_DATA = "correlation_input";
+
+  public CorrelationAssayMetadataPlugin(@NotNull PluginContext<CorrelationPluginRequest, CorrelationComputeConfig> context) {
+    super(context);
+  }
+
+  @NotNull
+  @Override
+  public List<StreamSpec> getStreamSpecs() {
+    // Get the collection variable and its entity
+    CorrelationComputeConfig computeConfig = getConfig();
+    CollectionSpec collectionVariable1 = computeConfig.getCollectionVariable1();
+    String entityId = collectionVariable1.getEntityId();
+
+    // Wrangle into correct types for what follows
+    EntityDef entity = getContext().getReferenceMetadata().getEntity(entityId).orElseThrow();
+    VariableSpec collectionVariable1VarSpec = VariableDef.newVariableSpec(entityId, collectionVariable1.getCollectionId());
+
+
+    // Grab all continuous variabls from ancestors
+    // The next line only grabs the second-up ancestor instead of all of them, because when i grabbed them all
+    // the variables wold be duplicated in the output. I tried distinct() but that didn't work. Any ideas?
+    // I also noticed we were receiving stable ids in the oriringal request, so i've filtered them out below. There 
+    // is probably a nicer way to do this.
+    // This at least works for now!
+    Stream<VariableDef> descendantVariableStream = getContext().getReferenceMetadata().getAncestors(entity).get(1).getVariables().stream();
```

Comment:
trying to be sure i understand.. you are temporarily taking the root-most entity you are confident exists in the mbio data diagram (which i guess will be either participant or participant repeated measures depending on the study), and asking for its variables? how are you getting variables from its child entities? or are we not yet?

also, lets assume in some future we have vars on child entities or the current entity. what do we want to do about those? also pick up any of those if they have a 1:1 relationship w the current entity? 

do we need to provide some sort of messaging around how many vars we found and ignored, etc? also, do we think well need this other places? should we be talking about a helper somewhere outside this plugin? maybe one that takes a predicate?

Reply:
https://github.com/VEuPathDB/service-eda-compute/issues/72

Reply:
We're grabbing the grandparent of the assay entity currently which usually has the most continuous metadata vars. When i asked for variables from all of the ancestors, i got duplicates which were complicated to de-duplicate. Since the goal of this PR is only to get something functional, i limited the scope and only now get vars from one ancestor entity. The linked issue #72  is marked as a must have so that we'll ensure we fix this issue, but this PR can be separate. 

Regarding the future where we might have child entities, yes i 'm imagining we allow these as long as they're 1:1. I don't know of any examples of that situation yet.

Reply:
duplication implies the child entities inherit the parent variables no? shouldnt we be able to pick the entity were building the plot in then? what am i not getting?

Reply:
moving this discussion to the linked issue...